src/net.c

```diff
@@ -2112,6 +2116,12 @@ coap_dispatch(coap_context_t *context, coap_session_t *session,
   else if (COAP_PDU_IS_RESPONSE(pdu))
     handle_response(context, session, sent ? sent->pdu : NULL, pdu);
   else {
+    if (COAP_PDU_IS_EMPTY(pdu)) {
```

Comment:
Should the ping handler also be called when a COAP_SIGNALING_PING message is received ? (line 2007)

Reply:
Good catch.  Yes, the ping handler should also be called when there is a detection of COAP_SIGNALING_PING - I will update the code.